packages/template-retail-react-app/app/constants.js

```diff
@@ -248,3 +252,10 @@ export const RESET_PASSWORD_LANDING_PATH = '/reset-password-landing'
 
 // Constants for Passwordless Login
 export const PASSWORDLESS_LOGIN_LANDING_PATH = '/passwordless-login-landing'
+
+export const PASSWORDLESS_ERROR_MESSAGES = [
+    /callback_uri doesn't match/i,
+    /error getting user info/i,
+    /passwordless permissions error/i,
+    /client secret is not provided/i,
```

Comment:
Where do we based these messages on? I assume this is from API. What happens if the API adds/removes these messages?

Reply:
Yes these messages are from the API. If the error message does not match any of these, the error will show the generic `API_ERROR_MESSAGE`, which is "Something went wrong! Please try again."

Reply:
my main concern is to have to maintain this list of constant. In a case where they change a bit of wording and an error will become obsolete.

Reply:
Looking at the [spreadsheet of error states](https://salesforce.quip.com/97bPANYv5D2U) I wonder if instead of matching by error message regex we should base it on status code? It seems like it would be possible